src/main/msp/msp.c

```diff
@@ -2841,7 +2841,7 @@ static mspResult_e mspProcessInCommand(mspDescriptor_t srcDesc, uint8_t cmdMSP,
                 mspArmingDisableByDescriptor(srcDesc);
                 setArmingDisabled(ARMING_DISABLED_MSP);
                 if (ARMING_FLAG(ARMED)) {
-                    disarm();
+                    disarm(DISARM_REASON_SERIAL_IO);
```

Comment:
I'd probably call this one `DISARM_REASON_ARMING_DISABLED` and the one below `DISARM_REASON_BST` - they are really different, with this one being a 'side effect' of arming being disabled, and the BST one being a 'disarm' command being received.

Reply:
perhaps `...SERIAL_IO` should be `DISARM_REASON_SERIAL_COMMAND`?

a disarm reason of arming disabled doesn't make much sense to me.

BST to me, is just another serial protocol like MSP and I didn't want to go down the route of specifically naming any serial protocol in the list of reasons.

Reply:
I'd still want to use different reasons for the two - the  situations are entirely different, and a disarm as a side effect of `MSP_SET_ARMING_DISABLED` is different from disarming because somebody explicitly wanted to disarm with `BST_DISARM`. We might even add a `MSP_DISARM` command at some point - there just does not seem to be need for this at the moment. Or there might be another way to set arming disabled that can occur when we are armed, requiring a disarm to happen as a side effect.
So maybe `DISARM_REASON_ARMING_DISABLED` and `DISARM_REASON_SERIAL_COMMAND`?`